### DISCUSSION TITLE
repro stuff - RANGES and beyond

### DISCUSSION THREAD

**Discussion by dustymc** [ArctosDB/arctos#8044] [General]
I'm not using the template because this is exploratory, not a proposal. Someone will need to refile if it becomes more.

Should have have a 'reproductive {class?? state?? something...}' attribute with values....

* reproductive (from https://github.com/ArctosDB/arctos/issues/6569 
* nipple enlarged - https://github.com/ArctosDB/arctos/issues/6595
* not nipple enlarged - https://github.com/ArctosDB/arctos/issues/6595
* vaginal perforation - https://github.com/ArctosDB/arctos/issues/6594
* not vaginal perforation - https://github.com/ArctosDB/arctos/issues/6594
* nulliparous, https://github.com/ArctosDB/arctos/issues/6593
* primiparous,  https://github.com/ArctosDB/arctos/issues/6593
* multiparous,  https://github.com/ArctosDB/arctos/issues/6593
* parous https://github.com/ArctosDB/arctos/issues/6593
* pregnancy https://github.com/ArctosDB/arctos/issues/6409
* not pregnancy.  https://github.com/ArctosDB/arctos/issues/6409
* placental scar (hard to fit this here) - https://github.com/ArctosDB/arctos/issues/6592
* lactation https://github.com/ArctosDB/arctos/issues/6408
* not lactation https://github.com/ArctosDB/arctos/issues/6408
* testis width (no fit?) https://github.com/ArctosDB/arctos/issues/6407
* embryo count  (?) https://github.com/ArctosDB/arctos/issues/6412
* testis length (?) https://github.com/ArctosDB/code-table-work/issues/2
* https://arctos.database.museum/info/ctDocumentation.cfm?table=ctsex_cde ??!
* maybe some more stuff from https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type
* whatever else I missed

...?

Seems like a mostly-manageable code table, would provide convenient one-ish-stop shopping for anyone looking for categorical repro stuff. 

Some of the above probably won't actually fit.

Somehow grouping these things would probably make the data more accessible to researchers, and harder to get lost in for enter-ers.

I can't quite see how to form a 'number+units repro stuff" attribute, so we may be stuck with ~dozens of '.. length'-type things - I'm not sure how, or if, that should influence this.

Some of this could instead be values in https://arctos.database.museum/info/ctDocumentation.cfm?table=ctexamined_detected - eg "pregnancy" --> "[not] detected".

These could also be implemented as proposed, each individual attributes, many with their own code tables. 

See also https://github.com/ArctosDB/arctos/discussions/6179

@bryansmclean @Jegelewicz

**Comment by Jegelewicz** [comment]
Other stuff from attributes:

brood patch [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#brood_patch)
gonad [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#gonad)
gonad weight [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#gonad_weight)
left gonad length [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#left_gonad_length)
left gonad width [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#left_gonad_width)
reproductive data [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#reproductive_data)
right gonad length [ link ] 
right gonad width [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#right_gonad_width)

**Comment by bryansmclean** [comment]
I think a dedicated repro table gets back to the original request (prior to RANGES), and is more attractive as a one-stop for data entry. We have a few additional straggler attributes that Im still working on requests for, Ill try and add them to this thread once I do that.

**Comment by bryansmclean** [comment]
> Some of the above probably won't actually fit.

This may be the last consideration. Which of the above (or which categories of things above) do you mean?

**Comment by dustymc** [comment]
>Which of the above

Placental scars for example - it really needs numbers (and maybe direction) - it's not clear to me how it might work as a categorical, which might mean

* one attribute for categorical repro stuff
* reproductive data [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#reproductive_data) for free-text repro stuff
* a whole bunch of number+units repro-related measurements and counts

And perhaps someone should propose merging some of the free-text stuff - eg gonad [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#gonad) (what a useful definition....) and brood patch [[ link ]](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type#brood_patch) are also free-text repro-related attributes, do we really need all three? If not they could be merged, if so the definitions could be updated to make that clear.

**Comment by bryansmclean** [comment]
> * placental scar (hard to fit this here) - [Code Table Request - placental scar count [RANGES project] #6592](https://github.com/ArctosDB/arctos/issues/6592)

This should be a case of number+units. And yes direction too.

ArctosDB/arctos#6588 on the other hand is to capture instances where placental scars were observed, but no count taken. Could just include "placental scarring" and "not placental scarring" in your proposed repro table to fix this

**Comment by ewommack** [comment]
https://github.com/ArctosDB/arctos/issues/5264

**Comment by Jegelewicz** [comment]
A question from today's meeting - should we add "repro: " to all reproductive attributes to cluster them together in the UI?

**Comment by dustymc** [comment]
https://docs.google.com/document/d/1HUXzIXs2qcG-BiCIJ33IXeW6lmOnT1U1U5Xvl7ZbtXo/edit - discussed, individual attributes are better, killing this.

**Comment by bryansmclean** [comment]
A thought on organizing growing lists of attributes, and easing the data entry process with that growing list (after yesterday's code table meeting): Could the prefix include a sex modifier? e.g.

**repro:female:placental scars**
**repro:male:testis position**

**Comment by bryansmclean** [comment]
Compiling here the list of new repro: attribute requests for RANGES, as outlined in AWG and code table discussions this week. Individual issues also updated to the best of my ability.

* **repro:vaginal perforation** https://github.com/ArctosDB/arctos/issues/6594
* **repro:parity** https://github.com/ArctosDB/arctos/issues/6593
* **repro:placental scars** https://github.com/ArctosDB/arctos/issues/6588
* **repro:placental scar count total** https://github.com/ArctosDB/arctos/issues/6592
* **repro:placental scar count left** https://github.com/ArctosDB/arctos/issues/6592
* **repro:placental scar count right** https://github.com/ArctosDB/arctos/issues/6592
* **repro:pregnant** https://github.com/ArctosDB/arctos/issues/6409
* **repro:embryo count total** https://github.com/ArctosDB/arctos/issues/6412
* **repro:embryo count left** https://github.com/ArctosDB/arctos/issues/6412
* **repro:embryo count right** https://github.com/ArctosDB/arctos/issues/6412
* **repro:nipple enlargement** https://github.com/ArctosDB/arctos/issues/6595
* **repro:lactating** https://github.com/ArctosDB/arctos/issues/6408
* **repro:testis position** https://github.com/ArctosDB/code-table-work/issues/3
* **repro:testis width** https://github.com/ArctosDB/arctos/issues/6407
* **repro:testis width left** https://github.com/ArctosDB/arctos/issues/6407
* **repro:testis width right** https://github.com/ArctosDB/arctos/issues/6407
* **repro:testis length** https://github.com/ArctosDB/code-table-work/issues/2
* **repro:testis length left** https://github.com/ArctosDB/code-table-work/issues/2
* **repro:testis length right** https://github.com/ArctosDB/code-table-work/issues/2

**Comment by bryansmclean** [comment]
Also starting a thread for existing attributes that need to receive **repro:** prefix. 
Here are the remaining ones within **Mamm:**

repro:[reproductive data](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type&coln=Mamm#reproductive_data)
repro:[right gonad length](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type&coln=Mamm#right_gonad_length)
repro:[left gonad length](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type&coln=Mamm#left_gonad_length)
repro:[right gonad width](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type&coln=Mamm#right_gonad_width)
repro:[left gonad width](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type&coln=Mamm#left_gonad_width)

**Comment by jldunnum** [comment]
Are pregnancy, not pregnancy, lactation and not lactation the proper terms? Seems like these could be the present condition or a prior condition. Pregnant and lactating would be correct for the condition at time of examination.

______________________________________________________________
Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131

**Comment by bryansmclean** [comment]
Good points and I am ok with this change (and will edit the post above)

**Comment by cjconroy** [comment]
Postpartum?  
I think that we are going to need a guidebook of sorts with many example. E.g. , If your data say this, this is how to input it into arctos. At the moment, I am not sure where to start other than parsing out all the details at MSB has started, and hoping they will fit into arctos fields.

**Comment by bryansmclean** [comment]
* **repro:parity** https://github.com/ArctosDB/arctos/issues/6593
shouldnt this cover "postpartum" if "parous" is selected for the above attribute?

**Comment by cjconroy** [comment]
I agree. But, what about having pregnant in that field as well? Instead of a separate field for pregnancy, have one field for all possibilities: never pregnant (nulliparous), currently pregnant, and the other categories of previously pregnant?

And this might sound like it would cause problems, but would it be possible to automatically fill in "pregnant" if there is a value in embryos, parous if there are scars, etc.?  I think for those of us preparing data for upload, we will possibly due this in our spreadsheets to help populate a separate pregnant field.

**Comment by dustymc** [comment]
>having pregnant in that field as well

That's starting to circle around to the first post, a "categorical repro stuff" attribute. Nobody seems to have any ideas about pulling number+units or freetext in there, so that'd leave something like....

reproductive
not reproductive
nipple enlarged
nipple not enlarged
vaginal perforation
no vaginal perforation
nulliparous
primiparous
multiparous
parous
pregnant
not pregnant
lactating
not lactating

(And I can't quite put my finger on why the group thought individual attributes was a better approach, but I think they did.)

>possible to automatically fill in "pregnant" if there is a value in embryos

we could perhaps set up a bot or something, maybe whatever gets worked out there would also be somehow useful for things like age/age class/numeric age. Someone's eventually going to have to write an "ontologies meet reality" proposal....

**Comment by bryansmclean** [comment]
> have one field for all possibilities

pregnancy and (multi)parity are possible simultaneously, which would mean youd select the same attribute twice with different values under the scheme we have now. i know this isnt a roadblock, just pointing it out.

> > possible to automatically fill in "pregnant" if there is a value in embryos
> 
> we could perhaps set up a bot or something, maybe whatever gets worked out there would also be somehow useful for things like age/age class/numeric age. Someone's eventually going to have to write an "ontologies meet reality" proposal....

For parsing traits from messy fields in which they currently get dumped (dwc:dynamicProps), pregnant=Y might be preferable to pregnant/non(or not) pregnant. 

The bot idea scares me a little because you are adding traits that arent actually recorded, but inferred. Its easy enough in my opinion to assume pregnancy if an embryo count is there.

**Comment by dustymc** [comment]
>bot idea scares me a little

It probably should, but I also think we have to get away from the viewpoint that any of this is anything like "facts" and metadata can be ignored. (We keep circling around to the idea that even the most straightforward things become ambiguous under close scrutiny if there's no metadata involved!) Part of that metadata involves the Agent making the assertion, which for a bot would be something like https://arctos.database.museum/agent/21344957 - a specific type of an Agent with a 'it's a script, this thing isn't capable of making independent assertions, it's just helped you find something and now you need to go check upstream data to determine if that's all lies or not" remark.

(And https://github.com/ArctosDB/arctos/discussions/5310 clearly needs to consider more than GUIDs - "reciprocal_relationship_bot" maybe hints at something, but https://arctos.database.museum/agent/21344957 is completely unambiguous - same as MVZ:Mamm:27928 vs http://arctos.database.museum/guid/MVZ:Mamm:27928. We can and should fully participate in the Extended Specimen Network!)

**Comment by Jegelewicz** [comment]
The addition of directions and methods (left, length) makes "method" easy to ignore. So much of the legacy data will not include a determiner, either because we really don't know who recorded the trait or we know, but we don't know the person well enough to create an unambiguous agent. For these reasons I think we have mostly ignored "metadata" in attributes but I think it is time to get more serious about using method and determiner and using them well? Could we build some sort of "method builder" tool that helps create standardized methods? Pick one each of anatomical direction, measurement type (length etc), and then free text for anything else you want to add. Creates a standard format for method and eliminates "L" vs "left" and so on as well as the need for 6 attributes for every measurement.

**Comment by bryansmclean** [comment]
I am circling back to this so we maintain momentum for reaching consensus. @cjconroy raises the point of lumping pregnancy (Y/N) in with parity. This seems to make sense, except that the categories for **parity** can be past or present tense. e.g. **primiparous** can mean "pregnant for the first time" or "having been pregnant for the first time". [https://en.wiktionary.org/wiki/primiparous]. **Pregnant** is always in the present tense. What say you @jldunnum @ewommack?

**Comment by jldunnum** [comment]
I kinda think its different in that pregnancy is a definitive observation based on the presence of embryos. Parity data is somewhat subjective because there is a lot of room for error. Just because you didn't see scars doesn't mean it definitively hasn't been pregnant before. Often quite hard to see scars on a very pregnant animal for example.
I guess my thought is that we will likely continue to just enter the straightforward data (i.e. embryos, scars, perforated vagina, etc.) and users can take those data and interpret them as they will.  It also means adding data entry; you will now have "pregnant" plus "primiparous" or "multiparous".

______________________________________________________________

Jonathan L. Dunnum Ph.D. (he, him, his)
Senior Collection Manager
Division of Mammals, Museum of Southwestern Biology
University of New Mexico
Albuquerque, NM 87131
(505) 277-9262
Fax (505) 277-1351

Chair, Systematic Collections Committee, American Society of Mammalogists
Latin American Fellowship Committee, ASM

MSB Mammals website: http://www.msb.unm.edu/mammals/index.html
Facebook: http://www.facebook.com/MSBDivisionofMammals

Shipping Address:
Museum of Southwestern Biology
Division of Mammals
University of New Mexico
CERIA Bldg 83, Room 204
Albuquerque, NM 87131
________________________________

**Comment by dustymc** [comment]
>pregnancy is a definitive observation based on the presence of embryos

NO! There are about a brazillion ways that could be inferred (some of them maybe even accurate); method is critical.

>interpret them as they will

Withholding method doesn't leave a lot of options!

**Comment by bryansmclean** [comment]
Philosophically I totally agree with the need to be specific with method (and other attribute metadata). Will be critical in our larger data universe to make  more and more digital attributes usable.

However, I'll come back to my assertion that the suite of mammal measurements we are talking about in this thread is relatively small and highly standardized. There simply _aren't_ that many ways that mammalogists infer pregnancy in dead mammals. **Ear from crown** and **ear from notch** are problematic as discussed, but are outliers. 

For the sake of progress for the community and RANGES, can we move ahead as planned? I would like to see more comprehensive attribute metadata tracking but would also argue we should apply this to cases where more interesting and variable metadata actually exist (eg pathogen/viral screening/detection).

**Comment by bryansmclean** [comment]
Any additional thoughts on moving forward with these requests? What can RANGES do to help clarify or coordinate?

**Comment by Jegelewicz** [comment]
> can we move ahead as planned?

I am unclear as to the agreed-upon plan and it seems like there is still uncertainty?

https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1690499952
https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1684027385
https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1684131977
https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1684025552


Before we add or change any of these, I want to make sure that you guys are sure about what you want and that I understand it!

**Comment by bryansmclean** [comment]
> I kinda think its different in that pregnancy is a definitive observation based on the presence of embryos. Parity data is somewhat subjective because there is a lot of room for error. Just because you didn't see scars doesn't mean it definitively hasn't been pregnant before. Often quite hard to see scars on a very pregnant animal for example. I guess my thought is that we will likely continue to just enter the straightforward data (i.e. embryos, scars, perforated vagina, etc.) and users can take those data and interpret them as they will. It also means adding data entry; you will now have "pregnant" plus "primiparous" or "multiparous".

I agree with Jon to keep [pregnancy](https://github.com/ArctosDB/arctos/issues/6409) as a separate attribute from [parity](https://github.com/ArctosDB/arctos/issues/6593). @cjconroy raises a good point about the continuity of these two traits re:female reproductive status, but I do think that parity is much more nuanced to infer while pregnancy is often directly observed by mammalogists doing necropsies. @cjconroy how do you feel about this issue?

**Comment by bryansmclean** [comment]
> > can we move ahead as planned?
> 
> I am unclear as to the agreed-upon plan and it seems like there is still uncertainty?
> 
> [#6599 (comment)](https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1690499952) [#6599 (comment)](https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1684027385) [#6599 (comment)](https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1684131977) [#6599 (comment)](https://github.com/ArctosDB/arctos/issues/6599#issuecomment-1684025552)
> 
> Before we add or change any of these, I want to make sure that you guys are sure about what you want and that I understand it!

As far as I can see, these reference the same one issue (which we can work to clear up ASAP). The second link you posted doesnt seem to point to an actual issue, but something we agreed to previously. I can post as a separate issue if needed.

**Comment by dustymc** [comment]
>moving forward with these requests?

I'm still very unclear on the big picture. 

1. We seem to end up at the idea of more-consolidated attributes to the extent realistically possible (eg, one 'categorical repro stuff' attribute - this issue) over and over again, but then we always seem to wander off. If a decision has been made, can that be spelled out somewhere (ideally in some way that'll help guide a post-RANGES future)?
2. The values, definitions, and documentation all needs to be designed to scale. (And maybe that needs to somehow be a more-formal part of attributes or authorities in general or ?? documentation.) There are about a bajillion ways to assess pregnancy for example, if we go forward with assumptions about dead mice and mammalogists then just about inevitably
   * we'll be back here in a few years, and 
   * any data recorded under those assumptions will be a whole lot less useful than they should be
3. I'm totally lost on the shape of the attribute type - is there some desire to establish standards regarding 'prefixes' or to do that for some (new stuff only?) or to (eventually) do something more formal with https://github.com/ArctosDB/arctos/issues/6413 or keep doing what we've [always done](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctattribute_type) (arbitrary values) or ????

Totally possible that I'm the only one lost on any of that, but if I'm not then maybe some sort of a (brief) roadmap document would be useful?

**Comment by bryansmclean** [comment]
> 3\. is there some desire to establish standards regarding 'prefixes'

I will chime in with a response to ArctosDB/arctos#3, which seems easiest to answer right now. **YES** - establishing a **repro:** prefix is what we all decided. This seems scalable in that different (future) projects can utilize the prefix, or bin other attributes of interest by other prefixes.

**Comment by Jegelewicz** [comment]
I think we have decided to leave off prefixes. Tentatively closing.

**Comment by campmlc** [comment]
Discussed again in Code Table meeting today, with potential path forward if we can clear up communication. @bryansmclean can we schedule a meeting to discuss?
